bin/git-team

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+# Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+# Copyright [2016-2022] EMBL-European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Update repositories - non-Production repos are maintained as
+# shallow copies, to reduce file size and make pulls quicker.
+
+
+# Script to use to save space on target by "not" cloning "shallow" repo with full history and depth=1
+# Parameters:
+#   - `-o` git operation to process - default to `pull`
+#       - For operation other than `pull` shallow repository are deleted and re-cloned to target `branch`
+#       - op can be 'clone', 'checkout', or 'pull';
+#         NB: for 'pull', the branches parameter is not used.
+#   - `-c` repository category to process - team|shallow - default to `shallow`
+#   - `-b` branch to apply operation onto - no effect on `pull` operation - default to `main`
+#   - `-f` fall back branch if `branch` doesn't exists - no effect on `pull` operation - default to `main`
+#   - `-s` shallow repositories to process
+#   - `-p` production repositories to process
+#   - `-p|-s` can be either of:
+#       - "Ensembl" xxx repo name - ex: ensembl-production
+#       - "Git Ensembl config" as define in `./git-ensembl` command tool - ex: production
+#       - "File path" containing a list of repositories to process - one per line
+
+function usage {
+  echo "Usage: $0 [-o git_operation] [-c shallow|team] [-b branch] [-f default_branch] [-t target_dir] [-s shallow_file|repo|config] [-p team_file|repo|team_config]"
+  echo "Example: $0 -o clone -c shallow -s ~/shallow.repo"
+  exit 1
+}
+shallow=()
+prod=()
+while getopts 'o:c:s:p:t:b:f:d' opt; do
+  case ${opt} in
```

Comment:
Can we call `usage` if any required option are missing while running `git team` please?

Reply:
Last commit should have mandatory/consistency parameter check:
- if shallow: test that we actually have a shallow to proceed
- if team: test that we have shallow and team set. 
- Updated usage to make it clearer, I hope.

Reply:
Can you resolve this or do you think it needs further tweaking?